usaspending_api/reporting/v2/views/agencies/publish_dates.py

```diff
@@ -0,0 +1,147 @@
+                        Cast("published_date", output_field=TextField()),
+                        Value('", "certification_date": "'),
+                        Cast("certified_date", output_field=TextField()),
+                        Value('"}, "quarterly": "'),
+                        Cast("quarter_format_flag", output_field=TextField()),
+                        Value('"}'),
+                    )
+                ),
+            )
+            .exclude(agency_name__isnull=True)
+            .values("agency_name", "abbreviation", "toptier_code", "current_total_budget_authority_amount", "periods")
+        )
+        return self.format_results(results)
+
+    def format_results(self, result_list):
+        results = []
+        for result in result_list:
+            results.append(
+                {
+                    "agency_name": result["agency_name"],
+                    "abbreviation": result["abbreviation"],
+                    "agency_code": result["toptier_code"],
+                    "current_total_budget_authority_amount": result["current_total_budget_authority_amount"],
+                    "periods": sorted(
+                        [json.loads(x) for x in result["periods"]], key=lambda x: x["reporting_fiscal_period"]
+                    )
+                    if result.get("periods")
+                    else None,
+                }
+            )
+        return results
+
+    def get(self, request):
+        if "publication_date" in self.pagination.sort_key:
+            self.validate_publication_sort(self.pagination.sort_key)
+            sort_key = deepcopy(self.pagination.sort_key)
+            pub_sort = sort_key.split(",")[1]
+            self.pagination.sort_key = "publication_date"
+            results = sorted(
+                self.get_agency_data(),
+                key=lambda x: x["periods"][
+                    next(
+                        (index for (index, d) in enumerate(x["periods"]) if d["reporting_fiscal_period"] == pub_sort), 0
+                    )
+                ]["submission_dates"]["publication_date"],
+                reverse=(self.pagination.sort_order == "desc"),
+            )
+        else:
+            results = sorted(
+                self.get_agency_data(),
+                key=lambda x: x[self.pagination.sort_key],
+                reverse=(self.pagination.sort_order == "desc"),
+            )
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        results = results[self.pagination.lower_limit : self.pagination.upper_limit]
+        return Response(
+            {"page_metadata": page_metadata, "results": results, "messages": self.standard_response_messages}
+        )
+
+    @cached_property
+    def pagination(self):
+        sortable_columns = [
+            "agency_name",
+            "abbreviation",
+            "agency_code",
+            "current_total_budget_authority_amount",
```

Comment:
Contract shows "name" and "code" as sortable; does this code or the contract need to change?

Reply:
My mistake; old version

Reply:
The contract lists the fields as `agency_code` and `agency_name`